split-webapp/split/src/components/AuthorizedRoute.js

```diff
@@ -0,0 +1,14 @@
+import { Redirect, Route } from "react-router-dom";
+import React, { useContext } from "react";
+import { UserContext } from "../context/UserContext";
+
+const AuthorizedRoute = ({ path, component, exact }) =>
+  useContext(UserContext).username ? (
+    <Route path={path} component={component} exact={exact} />
+  ) : (
+    <Redirect to="/" />
```

Comment:
I think this causes a redirection regardless of where you are - e.g. if you go to the signup page and hit refresh, this causes the page to be redirected to the login page, even though the signup route is not an `AuthorizedRoute`. Is this intended?

Reply:
Ah yeah, oops, I think I tested the sign in page by clicking on the link versus testing the route that should redirect by putting it into my browser. Fix should be pretty simple and is incoming.